<launch>
  <arg name="engage_transition_manager_param" default="$(find-pkg-share engage_transition_manager)/config/engage_transition_manager.param.yaml"/>
  <arg name="vehicle_info_param_file" default="$(find-pkg-share vehicle_info_util)/config/vehicle_info.param.yaml"/>

  <!-- engage_transition_manager -->
  <node pkg="engage_transition_manager" exec="engage_transition_manager_exe" name="engage_transition_manager" output="screen">
    <!-- args -->
    <param from="$(var vehicle_info_param_file)"/>
    <param from="$(var engage_transition_manager_param)"/>

    <!-- rename input topics -->
    <remap from="kinematics" to="/localization/kinematic_state"/>
    <remap from="steering" to="/vehicle/status/steering_status"/>
    <remap from="trajectory" to="/planning/scenario_planning/trajectory"/>
    <remap from="control_cmd" to="/control/command/control_cmd"/>
    <remap from="control_mode_report" to="/vehicle/status/control_mode"/>
    <remap from="gate_operation_mode" to="/control/vehicle_cmd_gate/operation_mode"/>
    <remap from="operation_mode_request" to="operation_mode_request"/> <!-- from system -->


    <!-- rename output topics -->
    <remap from="engage_state" to="engage_state"/>
    <remap from="control_mode_request" to="/control/control_mode_request"/> <!-- to vehicle -->
  </node>
</launch>
